agent/internal/client/host_methods.go

```diff
@@ -278,8 +278,8 @@ func (c *hostCommunicator) GetExpansions(ctx context.Context, taskData TaskData)
 }
 
 // Heartbeat sends a heartbeat to the API server. The server can respond with
-// an "abort" response. This function returns true if the agent should abort.
-func (c *hostCommunicator) Heartbeat(ctx context.Context, taskData TaskData) (bool, error) {
+// an "abort" response. Returning evergreen.TaskFailed signals the agent to abort the task.
```

Comment:
What about `evergreen.TaskConflict`?

Reply:
Fixed

Reply:
Usually, when I write a comment, I try to communicate high-level intent and behavior. I feel like the comment is overly specifying the implementation of the agent's abort process (which is external to this method) by saying that it must cancel the context and must continue "the agent loop". Instead, it would be more useful/maintainable to say what the heartbeat's return value actually signifies and let the agent deal with how that is handled. I would instead say how the agent should interpret a task conflict at a high level.

I'm assuming `evergreen.TaskConflict` means that the agent is no longer authorized to run this task (due to the task secret changing) - maybe phrase as something along these lines:
"Heartbeat will return a non-empty task status if the agent should no longer run the task. Returning TaskConflict means the agent is no longer authorized to run this task and should move on to the next available one"

Reply:
Sounds good - updated comment to be more high-level